packages/core/src/new-api/internal/execution/onchain-state-transitions.ts

```diff
@@ -619,3 +587,43 @@ async function _readEventArg(
     };
   }
 }
+
+async function checkTransactionComplete(
+  state: ExecutionEngineState,
+  next: { txHash: string; futureId: string; executionId: number },
+  successConstructor: (
+    receipt: ethers.providers.TransactionReceipt
+  ) => ExecutionSuccess | ExecutionFailure | TransactionMessage
+): Promise<OnchainStateTransitionContinue | OnchainStateTransitionPause> {
+  const currentTransaction = await state.chainDispatcher.getTransaction(
+    next.txHash
+  );
+
+  if (currentTransaction === null || currentTransaction === undefined) {
+    throw new IgnitionError(
```

Comment:
I don't think this is the intended behavior. We should be trying to discern if this is due to user interference or just dropped, and report different errors.

Reply:
I have added an issue for this one here: https://github.com/NomicFoundation/ignition/issues/313